### PR TITLE
RGB: Using 2D mappings for effects (+QoL changes)

### PR DESCRIPTION
**Pull request by henrygab**

### CODE REVIEW

File: pirate/rgb.c [hunk 1]
```diff
@@ -7,35 +8,146 @@
 #include "system_config.h"
 #include "pirate/rgb.h"
 
-#define RGB_MAX_BRIGHT 32
+
```
Comment (DangerousPrototypes) [inline pirate/rgb.c:11]: Love the ascii art!

File: pirate/rgb.c [hunk 2]
```diff
@@ -44,231 +156,173 @@ typedef struct _CPIXEL_COLOR_GRB {
...
+    //    each of which flows towards the plank connector (in opposite directions)
+    //
+    // groups_center_clockwise[]:
+    //    Similar to a clock, rotating around the device clockwise.
+    //
+    // groups_top_down[]:
+    //    All the pixels facing upwards as one group, and all the pixels
+    //    facing the sides as a second group.
+
+    #if BP5_REV <= 9
```
Comment (DangerousPrototypes) [inline pirate/rgb.c:216]: Thank you for cleaning this up, having it on a single line was a nightmare.

File: pirate/rgb.c [hunk 3]
```diff
@@ -509,31 +665,34 @@ bool rgb_timer_callback(struct repeating_timer *t){
...
             break;
         case LED_EFFECT_SCANNER:
+            //static const CPIXEL_COLOR scanner_color = { .r = 0x80, .g = 0x20, .b = 0x20 };
+            //next = animation_angular_wipe(scanner_color);
+            //next = animation_anticlockwise_scan(scanner_color);
             next = rgb_scanner();
             break;
         case LED_EFFECT_GENTLE_GLOW:
-            next = rgb_gentle_glow();
+            next = animation_gentle_glow();
```
Comment (DangerousPrototypes) [inline pirate/rgb.c:695]: I've been enjoying this new effect

**Comment by henrygab** [comment]
Although this does not include any gamma correction, I think this can be merged.

**Comment by DangerousPrototypes** [comment]
So sorry! I didn't see this changed from draft.

**Pull request merged**